src/hotspot/share/ci/ciKlass.cpp

```diff
@@ -75,12 +75,16 @@ bool ciKlass::is_subtype_of(ciKlass* that) {
     return true;
   }
 
-  VM_ENTRY_MARK;
-  Klass* this_klass = get_Klass();
-  Klass* that_klass = that->get_Klass();
-  bool result = this_klass->is_subtype_of(that_klass);
+  bool is_subtype;
+  GUARDED_VM_ENTRY(is_subtype = get_Klass()->is_subtype_of(that->get_Klass());)
 
-  return result;
+#ifdef ASSERT
+  if (that->is_instance_klass() && !that->is_interface()) {
```

Comment:
Why the !that->is_interface() test?

Reply:
`ciInstanceKlass::has_subklass()` is always `false` for interfaces because they always have `Klass::_subklass == NULL`.